source/local/local.go

```diff
@@ -143,6 +183,9 @@ func getFeaturesFromHooks() (map[string]string, error) {
 		}
 		return features, fmt.Errorf("unable to access %v: %v", hookDir, err)
 	}
+	if !(len(files) > 0) {
+		printHooksWarning = false
+	}
```

Comment:
Hmm, now it looks that we only print the warning when NO hooks are found 🧐

I'd just print it here (without any ratelimiting)

```suggestion
	if !(len(files) > 0) {
		klog.Warning("hooks are DEPRECATED since v0.12.0 and support will be removed in a future release; use feature files instead")
		}
```

Reply:
@marquiz No, I would say it is the opposite
`if !(len(files) > 0)` -> when hooks dir is empty, don't print. Perhaps you missed exclamation mark

Reply:
That should really be written without '!' for clarity: `if len(files) == 0`.

Also, above warning is about *use* of a deprecated API.

Because enabling hooks opens potential NFD security hole [1], I think security warning should be shown when hooks are enabled.  For example:
`klog.Warning("insecure hooks API enabled with 'HooksEnabled: true' (DEPRECATED since v0.12.0)")`

[1] of NFD executing random binaries from host, which may be  insecure, or even potentially malicious, and hooks require having extra components in NFD container which may later turn out to contain security holes.

Reply:
I agree with your first point on simplifying condition. But for the warning message it seems we are overcomplicating it. There could be another issue apart from security. I think, if there are some drawbacks of using a feature it should be documented clearly but not necessarily warned in the log trace.

Reply:
> of NFD executing random binaries from host

I wouldn't say random binaries from host. System / cluster admin is responsible for those in a similar way that any "system" files, I'd say. If the administrator just randomly installs any scripts or runs random containers with root privileges nfd is one of the minor problems :)

Reply:
They are random from the **NFD** PoV, and may force keeping it less secure.  It does not e.g. know whether what they do matches NFD security context settings: can some of its security context settings tightened or would that break some random hook?

Reply:
I get that but nfd-worker (with the default deployments) runs with probably the tightest security context possible: fileystem and all mounts are ro, run-as-non-root, allow-privilege-escalation=false, all caps dropped...

Reply:
Ok. I concur that warning about deprecated API *use* is enough.  And while there's no rate-limiting, it's indeed better just to output that warning directly in getFeaturesFromHooks().   Adding package-global "printHooksWarning" flag just to do warning from calling function is redundant complication.

Reply:
@eero-t No problem. @marquiz I think it is good to merge it now.

Reply:
Yeah, looks good now